source/slang/hlsl.meta.slang

```diff
@@ -6573,6 +6573,94 @@ matrix<T, N, M> sqrt(matrix<T, N, M> x)
     MATRIX_MAP_UNARY(T, N, M, sqrt, x);
 }
 
+// The following cbrt(float) and rcbrt(float) algorithms are based upon:
+//
+// Moroz L, Samotyy V, Walczyk CJ, Cieśliński JL.
+// Fast Calculation of Cube and Inverse Cube Roots Using a Magic Constant and Its Implementation on Microcontrollers.
+// Energies. 2021; 14(4):1058.
+// https://doi.org/10.3390/en14041058
+//
+// Published under the CC BY 4.0 license.
+//
+// They use no lookup tables, which makes them vectorizable in theory.
+
+[__readNone]
```

Comment:
Is this faster than calling `pow(x, 0.3333333333f)`? It is a lot of instructions.

Reply:
The linked paper claims it is 20x faster than that using an unspecified C standard library, but I can benchmark this under Vulkan, which may have different results.

Reply:
`pow()` appears to run about 2 to 3 times faster in a Vulkan compute shader under a loop of `1_000_000ll` iterations. The only suspicious thing to me is that `pow<float>(buff[0], 0.3333333333f);` compiles in GLSL as `pow(buff_0._data[0U], 0.3333333432674407959);` and I'm not sure where that number comes from. Maybe the compiler is reading the text `0.3333333333f` and parsing it to a float with slight error, then writing that back into the output shader?